lib/service/Service.ts

```diff
@@ -133,72 +76,269 @@ class Service extends EventEmitter {
   /**
    * Get the list of the order book's available pairs.
    * @returns A list of available trading pairs
+   * @throws ServiceError with status.FAILED_PRECONDITION code
    */
   public getPairs = () => {
-    return this.orderBook.getPairs();
+    try {
+      return this.orderBook.getPairs();
+    } catch (err) {
+      const serviceError: ServiceError = {
+        code: status.FAILED_PRECONDITION,
+        metadata: err,
```

Comment:
`metadata` is actually supposed to be grpc.metadata object, so I'm not sure this works. Same for other cases where `metadata` is assigned a value. https://grpc.io/grpc/node/grpc.Metadata.html

Reply:
will check that ! :)